tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -142,13 +143,38 @@ static unique_ptr<QueryResult> CompletePendingQuery(PendingQueryResult &pending_
 	return pending_query.Execute();
 }
 
-DuckDBPyConnection *DuckDBPyConnection::Execute(const string &query, py::object params, bool many) {
+py::list TransformNamedParameters(string &query, py::dict params) {
+	py::list new_params(params.size());
+
+	idx_t param_idx = 1;
+	for (auto &item : params) {
+		const std::string &item_name = item.first.cast<std::string>();
+		auto param_name = StringUtil::Format("$%s", item_name);
+		if (query.find(param_name) == std::string::npos) {
+			throw InvalidInputException(
+			    "Named parameters could not be transformed, because query string is missing named parameter '%s'",
+			    param_name);
+		}
+		// Replace all occurences of $name with $<index>
+		query = StringUtil::Replace(query, param_name, StringUtil::Format("$%d", param_idx));
```

Comment:
Very much an edge case, but what if $variable appears inside a string in the query?

Reply:
Ah I thought about this but only in relation to column names, which I reasoned cant contain `$`, but you raise a valif point, string literals can contain this..

Definitely needs a test for this and probably requires changing the implementation 👍

Reply:
My main concern is for people using regex's, where `'$hello'` is perfectly valid regex

Reply:
Yes, definitely, already said you're right :+1:
Need to change the implementation

Reply:
Yep just giving a realish query example that could break